src/figuresClasses.ts

```diff
@@ -1,19 +1,76 @@
-export interface Figure {
+type FigureColor = 'red' | 'green' | 'blue';
+type FigureShape = 'triangle' | 'circle' | 'rectangle';
 
+export interface Figure {
+  shape: FigureShape;
+  color: FigureColor;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: FigureShape = 'triangle';
 
-}
+  constructor(
+    public color: FigureColor,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0
+      || b <= 0
+      || c <= 0
+      || (a + b) <= c
+      || (c + a) <= b
+      || (c + b) <= a
+    ) {
+      throw new Error('Wrong sides!');
+    }
+  }
 
-export class Circle {
+  getArea(): number {
+    const halfPerim = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(halfPerim * (halfPerim - this.a)
+      * (halfPerim - this.b) * (halfPerim - this.c));
```

Comment:
you can use destructuring for your values, but this is optional;

```suggestion
		{a, b , c} = this;
    const halfPerim = (a + b + c) / 2;
    const area = Math.sqrt(halfPerim * (halfPerim - a)
      * (halfPerim - b) * (halfPerim - c));
```